dogpile/cache/backends/memcached.py

```diff
@@ -509,15 +563,60 @@ def __init__(self, arguments):
         self.serde = arguments.get("serde", pymemcache.serde.pickle_serde)
         self.default_noreply = arguments.get("default_noreply", False)
         self.tls_context = arguments.get("tls_context", None)
+        self.socket_keepalive = arguments.get("socket_keepalive", None)
+        self.enable_retry_client = arguments.get("enable_retry_client", False)
+        self.retry_attempts = arguments.get("retry_attempts", None)
+        self.retry_delay = arguments.get("retry_delay", None)
+        self.retry_for = arguments.get("retry_for", None)
+        self.do_not_retry_for = arguments.get("do_not_retry_for", None)
+        if (self.retry_delay is not None or
+            self.retry_attempts is not None or
+            self.retry_for is not None or
+            self.do_not_retry_for is not None) and \
+           self.enable_retry_client is None:
+            raise ValueError(
+                "You try to configure pymemcache's retry "
+                "mechanisms without enabling them. You see this error "
+                "message because `enable_retry_client` is `False`. To "
+                "avoid this message please consider enabling the retry "
+                "mechanisms by passing `enable_retry_client` is `True`.")
 
     def _imports(self):
         global pymemcache
         import pymemcache
 
     def _create_client(self):
-        return pymemcache.client.hash.HashClient(
-            self.url,
-            serde=self.serde,
-            default_noreply=self.default_noreply,
-            tls_context=self.tls_context,
-        )
+        _kwargs = {
+            "serde": self.serde,
+            "default_noreply": self.default_noreply,
+            "tls_context": self.tls_context
+        }
+        if self.socket_keepalive is not None:
+            try:
+                # Check if pymemcache's keepalive features are supported
+                # by the installed version of pymemcache. They are supported
+                # since the version 3.5.0 of pymemcache. The import below
+                # will fail if the used version isn't compatible.
+                from pymemcache import KeepaliveOpts # noqa
```

Comment:
So, I updated my previous version to update the kwargs only if the args is provided and so avoid issues after upgrade. I also added tests to check if the pymemcache is compatible with the passed params, just in case if someone try using retry mechanisms and socket keepalive with a non compatible version.